tools/profile-endpoints.py

```diff
@@ -0,0 +1,355 @@
+    return None
+
+def authenticate():
+  """
+  Prompts the user to authenticate using gcloud auth login.
+  """
+  print("Please complete the authentication in the web browser that opens.")
+  auth_command = "gcloud auth login"
+  run_command(auth_command)
+
+def get_access_token():
+  """
+  Retrieves the access token using gcloud auth print-access-token.
+
+  Returns:
+      str: The access token if successful; otherwise, None.
+  """
+  token_command = "gcloud auth print-access-token"
+  return run_command(token_command)
+
+# HTTP Request Functions
+
+def make_get_request(endpoint_url, token):
+  """
+  Makes a GET request to the specified endpoint with the given access token.
+
+  Args:
+      endpoint_url (str): The endpoint URL to request.
+      token (str): The access token for authentication.
+
+  Returns:
+      float: The time taken for the request in seconds.
+  """
+  url = f"{DATAREPO_URL}/api/repository/v1/datasets/{UUID}/snapshotBuilder/{endpoint_url}"
+  headers = {
+    "Authorization": f"Bearer {token}"
+  }
+
+  try:
+    start_time = time.time()
+    response = requests.get(url, headers=headers)
+    end_time = time.time()
+
+    total_time = end_time - start_time
+
+    if response.status_code == 200:
+      print("Request successful!")
+    elif response.status_code == 502:
+      print("Request timed out, to get a more accurate time, please use the Logs Explorer at console.cloud.google.com for accurate timings")
+    else:
+      print(f"Request failed with status code {response.status_code}: {response.text}")
+
+    return total_time
+  except requests.exceptions.RequestException as e:
+    print(f"An error occurred during the request: {e}")
+    return None
+
+def make_post_request(endpoint_url, token, body):
+  """
+  Makes a POST request to the specified endpoint with the given access token and request body.
+
+  Args:
+      endpoint_url (str): The endpoint URL to request.
+      token (str): The access token for authentication.
+      body (dict): The request body to send in the POST request.
+
+  Returns:
+      tuple: A tuple containing the response object and the time taken for the request in seconds.
+  """
+  url = f"{DATAREPO_URL}/api/repository/v1/datasets/{UUID}/snapshotBuilder/{endpoint_url}"
+  headers = {
+    "Authorization": f"Bearer {token}",
+    "Content-Type": "application/json"  # Indicating the request body is in JSON format
+  }
+
+  try:
+    start_time = time.time()
+    response = requests.post(url, headers=headers, json=body)
+    end_time = time.time()
+
+    time_taken = end_time - start_time
+
+    if response.status_code == 200:
+      print("Request successful!")
+    elif response.status_code == 502:
+      print("Request timed out, to get a more accurate time, please use the Logs Explorer at console.cloud.google.com for accurate timings")
+    else:
+      print(f"Request failed with status code {response.status_code}: {response.text}")
+
+  return time_taken
```

Comment:
This code doesn't run in python. I was able to get it to work by indenting `return`
```suggestion
    return time_taken
```

Reply:
Sorry about this!!